### PR TITLE
DOC: update the day_name function docstring

### PR DESCRIPTION
**Pull request by NeehaK**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [ X] PR title is "DOC: update the <your-function-or-method> docstring"
- [ X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [ X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
Docstring for "pandas.DatetimeIndex.day_name" correct. :)
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/datetimes.py [hunk 1]
```diff
@@ -2128,19 +2128,34 @@ def month_name(self, locale=None):
...
-            Index of day names
-
-        .. versionadded:: 0.23.0
+        day_names : Index
+            Index (instance) of day names
+            
+        See Also
+        --------
+        month_name : Return the month names
+
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: Add `locale` which will link to https://docs.python.org/3/library/locale.html

File: pandas/core/indexes/datetimes.py [hunk 2]
```diff
@@ -2128,19 +2128,34 @@ def month_name(self, locale=None):
 
     def day_name(self, locale=None):
         """
-        Return the day names of the DateTimeIndex with specified locale.
+        Return the day names.
+        
+        Provides day names of the DateTimeIndex for the dataframe which consists of the time and date data with specified locale.
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: DatetimeIndex (lowercase t)

Comment (TomAugspurger) [reply pandas/core/indexes/datetimes.py]: This section can probably be removed.

File: pandas/core/indexes/datetimes.py [hunk 3]
```diff
@@ -2128,19 +2128,34 @@ def month_name(self, locale=None):
...
 
         Parameters
         ----------
         locale : string, default None (English locale)
-            locale determining the language in which to return the day name
-
+            The locale determines the language in which the day name should be returned.
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: This should go just below the `locale` line.

File: pandas/core/indexes/datetimes.py [hunk 4]
```diff
@@ -2128,19 +2128,34 @@ def month_name(self, locale=None):
...
     def day_name(self, locale=None):
         """
-        Return the day names of the DateTimeIndex with specified locale.
+        Return the day names.
+        
+        Provides day names of the DateTimeIndex for the dataframe which consists of the time and date data with specified locale.
 
         Parameters
         ----------
         locale : string, default None (English locale)
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]:
```
locale : string, optional
```

File: pandas/core/indexes/datetimes.py (hunk 3 above)
Comment (TomAugspurger) [reply pandas/core/indexes/datetimes.py]: Mention that the default is the current locale, which can be obtained with `locale.getlocale()`.

File: pandas/core/indexes/datetimes.py [hunk 5]
```diff
@@ -2128,19 +2128,34 @@ def month_name(self, locale=None):
...
+        day_names : Index
+            Index (instance) of day names
+            
+        See Also
+        --------
+        month_name : Return the month names
+        
+        Examples
+        --------
+        #From the time stamps data based on it the day_name is returned.
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: Do you need help making examples? This should be like

```python
>>> idx = pd.date_range('2017', periods=2)
>>> idx
DatetimeIndex(['2017-01-01', '2017-01-02'],
              dtype='datetime64[ns]', freq='D')
>>> idx.day_name()
>>> idx.day_name()
Index(['Sunday', 'Monday'], dtype='object')
```

Then show an example with a different locale like

```python
In [37]: idx.day_name('zh_hk')
Out[37]: Index(['週日', '週一'], dtype='object')
```

**Review by mroeschke — COMMENTED** [review]
You'll want to keep the `.. versionadded:: 0.23.0` note. This is a new method that will be available in the next release.

**Comment by datapythonista** [comment]
The code for `day_name` changed, and this docstring seems to be defined somewhere else. Closing this PR. Thanks for the contribution @NeehaK.

**Pull request closed without merging**